From 740a1d1f8a8a99d38fdb29b5d5245e567cc27a95 Mon Sep 17 00:00:00 2001
From: Joseph Kogut <joseph@balena.io>
Date: Tue, 31 Jan 2023 16:04:13 -0800
Subject: [PATCH] console: Fix a typo in the EFI warning list in gnu-efi

Some versions of gnu-efi have a typo, in which "EFI_WARN_UNKNOWN_GLYPH"
is accidentally "EFI_WARN_UNKOWN_GLYPH".  Work around that, so that we
can use the not-silly one in console.c's list of error and warning
messages.

This is a backport from devel for:

  commit 5f62b22ccd636d326b3229a2b196118701c6f3f7
  Author: Peter Jones <pjones at redhat.com>
  Date:   Mon Aug 26 16:12:05 2019 -0400

Signed-off-by: Joseph Kogut <joseph@balena.io>
Signed-off-by: Peter Jones <pjones at redhat.com>
Upstream: d230d02f990f02293736dca78b108f86c86d1bd0
Signed-off-by: Thomas Petazzoni <thomas.petazzoni at bootlin.com>
---
 include/console.h | 6 ++++++
 lib/console.c     | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/console.h b/include/console.h
index 5ab5d6b..cb46815 100644
--- a/include/console.h
+++ b/include/console.h
@@ -1,3 +1,9 @@
+#if !defined(EFI_WARN_UNKNOWN_GLYPH) && defined(EFI_WARN_UNKNOWN_GLYPH)
+#define EFI_WARN_UNKNOWN_GLYPH EFI_WARN_UNKOWN_GLYPH
+#elif !defined(EFI_WARN_UNKNOWN_GLYPH)
+#define EFI_WARN_UNKNOWN_GLYPH EFIWARN(1)
+#endif
+
 EFI_INPUT_KEY
 console_get_keystroke(void);
 int
diff --git a/lib/console.c b/lib/console.c
index b932a44..dc0f9e7 100644
--- a/lib/console.c
+++ b/lib/console.c
@@ -357,7 +357,7 @@ static struct {
 	{  EFI_SECURITY_VIOLATION,     L"Security Violation"},
 
 	// warnings
-	{  EFI_WARN_UNKNOWN_GLYPH,     L"Warning Unknown Glyph"},
+	{  EFI_WARN_UNKNOWN_GLYPH,      L"Warning Unknown Glyph"},
 	{  EFI_WARN_DELETE_FAILURE,    L"Warning Delete Failure"},
 	{  EFI_WARN_WRITE_FAILURE,     L"Warning Write Failure"},
 	{  EFI_WARN_BUFFER_TOO_SMALL,  L"Warning Buffer Too Small"},
